src/redux/checkin.test.js

```diff
@@ -301,35 +325,246 @@ describe('checkin: adding tasks', async (assert) => {
+              description: 'Push code to GitHub',
+              checkinId: checkin2Id,
+              userId: user2Id,
+            }),
+            createTestTask({
+              description: 'Review GitHub PRs',
+              checkinId: checkin2Id,
+              userId: owner.id,
+            }),
+          ],
+        }),
       ],
-      tasks: [
-        createTestTask({
-          description: 'Build Checkin App',
-          checkinId: checkin.id,
-          userId: owner.id,
+    }),
+  });
+});
+
+describe('checkin: adding blockers', async (assert) => {
+  const owner = createTestUser({ id: '1' });
+  const team = createTestTeam({ id: '1', ownerId: owner.id, users: [owner.id] });
+  const checkin = createTestCheckin({
+    id: '1',
+    userId: owner.id,
+    teamId: team.id,
+    createdAt: 1591044882455,
+  });
+  const user2Id = '2';
+  const team2Id = '2';
+  const checkin2Id = '2';
+  const checkin2CreatedAtTimestamp = 1591044882456;
+  const createBaseBlockersTestState = ({
+    users = {
+      [owner.id]: owner,
+    },
+    teams = {
+      [team.id]: team,
+    },
+    checkins = [checkin],
+  } = {}) =>
+    createTestState({
+      users,
+      teams,
+      checkins,
+    });
+  const getInitialBlockerActions = () => [
+    addUser({ id: owner.id }),
+    addTeam({ id: team.id, ownerId: owner.id }),
+    addCheckin({ id: checkin.id, userId: owner.id, teamId: team.id, createdAt: checkin.createdAt }),
+  ];
+
+  assert({
+    given: 'adding a blocker to an existing checkin by an existing user',
+    should: 'ad the blocker to the state',
+    actual: [
+      ...getInitialBlockerActions(),
+      addBlocker({ description: 'Life gets in the way', checkinId: checkin.id, userId: owner.id }),
+    ].reduce(reducer, reducer()),
+    expected: createBaseBlockersTestState({
+      checkins: [
+        createTestCheckin({
+          ...checkin,
+          blockers: [
+            createTestBlocker({
+              description: 'Life gets in the way',
+              checkinId: checkin.id,
+              userId: owner.id,
+            }),
+          ],
+        }),
+      ],
+    }),
+  });
+
+  assert({
+    given: 'adding a blocker for a non-existent user',
+    should: 'should not modify state',
+    actual: [
+      ...getInitialBlockerActions(),
+      addBlocker({ description: 'Life gets in the way', checkinId: checkin.id, userId: user2Id }),
+    ].reduce(reducer, reducer()),
+    expected: createBaseBlockersTestState(),
+  });
+
+  assert({
+    given: 'adding a task to a non-existent checkin',
+    should: 'should not modify state',
+    actual: [
+      ...getInitialBlockerActions(),
+      addBlocker({ description: 'Life gets in the way', checkinId: checkin2Id, userId: owner.id }),
+    ].reduce(reducer, reducer()),
+    expected: createBaseBlockersTestState(),
+  });
+
+  assert({
+    given: 'adding multiple tasks to existing checkins by existing users',
+    should: 'add the tasks to the state',
+    actual: [
```

Comment:
For complex setup, try the Arrange, Act, Assert structure. Arrange first, then act, then assert last. That will make your assertion and the intention of the test much more clear.

Reply:
I ended up combining the Arrange and Act portions since it seemed like separating them wouldn't be of much benefit.